### ISSUE TITLE
Several Errors related to macos version

### ISSUE THREAD

**Issue by cerupcat** [apple/swift-package-manager#8134]
I'm not sure if this is a bug or lack of knowledge, but I have a iOS project that is defined as a package. I want to generate documentation for the main target within that package.

At the top of the Package.swift we have:

```
    platforms: [
        .iOS(.v16),
    ],
```

However, when running:

`swift package --allow-writing-to-directory ./docs generate-documentation --target MainApp --disable-indexing --transform-for-static-hosting --hosting-base-path MainApp --output-path ./docs
`

We see many errors related to macos version differences despite not trying to build macos documentation.

`
the library 'X' requires macos 10.13, but depends on the product 'Y' which requires macos 10.14; consider changing the library 'X' to require macos 10.14 or later, or the product 'Y' to require macos 10.13 or earlier.
`

Is there a way to specify not to build documentation for macos? We would expect the "platforms" definition in the Package file to be used to determine this.

**Comment by dschaefer2** [comment]
Do you not get the same errors when building this package?

**Comment by cerupcat** [comment]
> Do you not get the same errors when building this package?

No, we don't get the same errors when just building our app using the Package.swift file.

**Comment by dschaefer2** [comment]
OK. Thanks. I need to learn what the generate-documentation command does that's different from a simple package resolution.

**Comment by diegopetrucci** [comment]
@dschaefer2 FWIW I get similar errors when building/testing packages from CLI, eg `swift test --package-path MyPackage`. They do not appear with Xcode, and just like OP I have constrained the supported platforms to iOS only in my packages.

You can trigger this by having a fresh project, creating a package, and making it dependent on eg TCA. `swift test` will fail, Xcode will not.
